### ISSUE TITLE
Calling Disconnect() fails with error "not connected"

### ISSUE THREAD

**Issue by brutella** [soypat/natiu-mqtt#2]
After connecting to a MQTT broker, I call `Disconnect()` and get the error "not connected". I then checked the implementation of the `Disconnect()` method.

```go
if !c.lastRx.IsZero() {
    return errors.New("not connected")
}
```

Shouldn't the if-statement be like `if c.lastRx.IsZero() { ... }`?

**Comment by brutella** [comment]
Thanks for pointing that out. For my use case, a simple client implementation, this library is exactly what I was looking for. I've also implemented the basic message sequences for the different QoS levels already and it works nicely.

**Comment by soypat** [comment]
@brutella Is your implementation public?

**Comment by brutella** [comment]
No, the client is integrated into a private repo. Are you interested in the implementation?

**Comment by soypat** [comment]
I can confirm I would be interested in the implementation if it is not too much of a hassle for you.
